.github/workflows/docs.yml

```diff
@@ -0,0 +1,45 @@
+name: "Publish TypeDoc"
+on:
+  push:
+    branches:
+      - "main"
+
+jobs:
+  publish-documentation:
+    permissions:
+      id-token: "write"
+      pages: "write"
+
+    environment:
+      name: "github-pages"
+      url: "${{ steps.deployment.outputs.page_url }}"
+
+    runs-on: "ubuntu-latest"
+    steps:
+      - id: "checkout"
+        name: "Check out Git repository"
+        uses: "actions/checkout@v4"
+
+      - id: "setup-node"
+        name: "Set up Node.js"
+        uses: "actions/setup-node@v4"
+
+      - id: "install-dependencies"
+        name: "Install Node.js dependencies"
+        run: |
+          set -euo pipefail
+          npm install
```

Comment:
Shouldn't that be npm ci just like the other scripts? Also no need for the pipe fail thing. Just keep it simple.